### DISCUSSION TITLE
@veramo/cli full functionalities

### DISCUSSION THREAD

**Discussion by tientaidev** [decentralized-identity/veramo#774] [Q&A] [answered]
HI, I am new to SSI and want to explore workflow of Veramo.
I have some questions.
1) Currently, with Veramo/cli, I can only create/delete DIDs, create credential, create presentation right ?
2) What extra configuration steps that @mirceanis mention in #657 (Answer number 2) to get full functionalities ? What functionalities that I can do besides the ones that I mention in question 1 ?
3) I want to use features like did update, selective content generation, issuer update, etc. (Picture below). Should I continue using Veramo/cli or write Node.js app with other packages from Veramolabs ?
![image](https://user-images.githubusercontent.com/43933775/145069763-f3cca578-29c3-42ea-8be9-948d629e518a.png)

**Comment by mirceanis** [comment, marked as answer]
You can use veramo CLI to execute any method that the veramo agent exposes. These methods can be executed as terminal commands, using the `veramo execute` command, or they can be called through OpenAPI, when the agent is started as a server using the `veramo server` command.
This is very useful if you are developing an application in a different stack than Node.js, because you can make use of existing javascript plugins and treat veramo as an external service.
Veramo CLI can also be used with packages and plugins from other projects, so you are not limited to the transitive dependencies of the `@veramo/cli` package.

If you are already developing a Node.js app, then it makes more sense to use Veramo as a library inside your app.
Take a look at [this tutorial](https://veramo.io/docs/node_tutorials/node_setup_identifiers) on how to get started with Veramo in node.

One thing that may not be obvious at first glance is that Veramo is not limited to the plugins that are provided here.
you can develop your own plugins and use them in your agents and also use plugins developed by other teams.
This is an important point as it also means that there is no single "full functionality". You decide what functionality it should have based on the plugins you install.

I'm not familiar with all the concepts laid out in your diagram, but I think that all the white blocks are possible using existing Veramo plugins (either CLI, or server, or as a library.. it is up to you how you use it).

Veramo is a new project and it is evolving, as there are lots of things to be improved and implemented.
We rely on people like you who use it to let us know how it can be improved or to directly contribute new features and fixes.
We look forward to your feedback and contributions!
